.circleci/config.yml

```
@@ -83,27 +125,24 @@ workflows:
               only:
                 - main
     jobs:
-      - ruby-2.3: *tag_filters
-      - ruby-2.4: *tag_filters
-      - ruby-2.5: *tag_filters
-      - ruby-2.6: *tag_filters
-      - ruby-2.7: *tag_filters
+      - test:
+          <<: *matrix_rubyversions
   build:
     jobs:
-      - ruby-2.3: *tag_filters
-      - ruby-2.4: *tag_filters
-      - ruby-2.5: *tag_filters
-      - ruby-2.6: *tag_filters
-      - ruby-2.7: *tag_filters
-      - publish:
+      - test:
+          <<: *filters_always
+          <<: *matrix_rubyversions
+      - build_artifacts:
+          <<: *filters_always
```

should this be using the publish filters? It doesn't seem like we need the artifacts unless we are publishing to GitHub?

```suggestion
          <<: * filters_publish
```

---

I think it's handy to have the built artifact, it allows us to build and provide gems for manual installations for "unofficial" releases, e.g. to test/cut a bug fix patch for a customer without having to create an official release.

---

hmm ok, perhaps we should build it with a different version then? It might be a bit weird to have version `x.x.x` published on ruby gems being different to the same version that we give a customer

---

Oh - we are building it, but not publishing it in the current implementation. You will see the artifact in the CI artifacts, but not on github or rubygems.